### DISCUSSION TITLE
AttributeError: 'ScrolledText' object has no attribute 'select_range'

### DISCUSSION THREAD

**Discussion by botkero** [israel-dryer/ttkbootstrap#381] [Q&A] [answered]
Hello,

I noticed that I get an error when I press CTRL + A in ScrolledText in ttkbootstrap, I have created two minimal snippets here.

No error:
``` py
import tkinter as tk
from tkinter.scrolledtext import ScrolledText

root = tk.Tk()
root.geometry("500x500")

text = ScrolledText(root)
text.pack(fill=tk.BOTH, expand=tk.YES)

root.mainloop()
```

Error, when pressing “CTRL + A”:
``` py
import ttkbootstrap as ttk
from tkinter.scrolledtext import ScrolledText

root = ttk.Window()
root.geometry("500x500")

text = ScrolledText(root)
text.pack(fill=ttk.BOTH, expand=ttk.YES)

root.mainloop()
```

Since no one has encountered this problem, I think it's probably me. So, what am I doing wrong?
I'm on Windows 10 using 3.11 Python.

``` shell
Exception in Tkinter callback
Traceback (most recent call last):
  File "D:\WinPython11\python-3.11.0.amd64\Lib\tkinter\__init__.py", line 1948, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "D:\WinPython11\python-3.11.0.amd64\Lib\site-packages\ttkbootstrap\window.py", line 109, in on_select_all
    widget.select_range(0, END)
    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'ScrolledText' object has no attribute 'select_range'
```

(facepalm)
I didn't see this page... it works now!
https://ttkbootstrap.readthedocs.io/en/latest/api/scrolled/scrolledtext/

``` py
import ttkbootstrap as ttk
from ttkbootstrap.scrolled import ScrolledText

root = ttk.Window()
root.geometry("500x500")

text = ScrolledText(root)
text.pack(fill=ttk.BOTH, expand=ttk.YES)

root.mainloop()
```

**Comment by SZY2077** [comment]
Hello！
Now I can give you an answer.
```py
import ttkbootstrap as ttk
from ttkbootstrap.scrolled import ScrolledText

#void
def select_all():
    text.tag_add("sel", "1.0", "end")

root = ttk.Window()
root.geometry("500x500")

text = ScrolledText(root)
text.pack(fill=ttk.BOTH, expand=ttk.YES)

text.bind("<Control-a>",select_all)

root.mainloop()
```

**Comment by SZY2077** [comment]
If it has wrong too.
you can try edit D:\WinPython11\python-3.11.0.amd64\Lib\site-packages\ttkbootstrap\window.py (edit lines109-112):
```py
try:
    widget.select_range(0, END)
    widget.icursor(END)
except:pass
```

**Comment by israel-dryer** [comment, marked as answer]
Good catch — this was an actual bug in ttkbootstrap, not on your end.

`ttk.Window` binds `<Control-a>` globally, and the old handler called `widget.select_range(0, END)`, which `Text` widgets don't have (`tkinter.scrolledtext.ScrolledText` subclasses `tkinter.Text`, so it crashed).

It was fixed in **v1.13.12** (PR #696): `on_select_all` now detects `Text` widgets and selects with `tag_add`/`mark_set`, falling back to `Entry` behavior otherwise. Upgrading resolves it for both the stdlib `ScrolledText` and `ttkbootstrap.scrolled.ScrolledText` — no per-widget binding or `site-packages` edits needed.
